.github/actions/yarn-nm-install/action.yml

```diff
@@ -0,0 +1,65 @@
+########################################################################################
+# "yarn install" composite action for yarn 3/4+ and "nodeLinker: node-modules"         #
+#--------------------------------------------------------------------------------------#
+# References:                                                                          #
+#   - bench: https://gist.github.com/belgattitude/0ecd26155b47e7be1be6163ecfbb0f0b     #
+########################################################################################
+
+name: 'Monorepo install (yarn)'
+description: 'Run yarn install with node_modules linker and cache enabled'
+inputs:
+  enable-corepack:
+    description: 'Enable corepack'
+    required: false
+    default: 'false'
+  cache-install-state:
+    description: 'Cache yarn install state (invalidated on lock changes)'
+    required: false
+    default: 'true'
+  cache-node-modules:
```

Comment:
Caching `**/node_modules` is heavy for action/cache. Lot of files... zstd compression/decompression can easily takes few minutes. It's also invalidated on lock changes and/or architecture change.